apache/zeppelin
# [ZEPPELIN-2593] Add storage settings to persist on run and commit

### What is this PR for?
adding settings to storage layer to let user choose to persist continuously or only on commits


### What type of PR is it?
Improvement | Feature

### Todos
* [x] - add notebook repo setting
* [x] - add persist on run & commit (checkpoint)
* [x] - add persist on commit (checkpoint) only

### What is the Jira issue?
[ZEPPELIN-2593](https://issues.apache.org/jira/browse/ZEPPELIN-2593)

### How should this be tested?
1. go to notebook repo menu and click on edit `Global settings` or do through env var `ZEPPELIN_NOTEBOOK_PERSISTENCE`
2. change value of `Note persistence` item to `Persist on note run and commits` or change value of `ZEPPELIN_NOTEBOOK_PERSISTENCE` into "run"
3. go change any note, although it will be changed in memory, it won't be persisted
4. to check above can reload notes in note list view -> will see original note
5. can save note either by `Run` or clicking `Commit` button in version control menu of note

### Screenshots (if appropriate)
<img width="1191" alt="screen shot 2017-07-10 at 3 02 02 am" src="https://user-images.githubusercontent.com/1642088/27996389-fd0a4a26-651b-11e7-91ed-ea22d135f875.png">



## Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no
